src/lib/yang/schema.lua

```diff
@@ -795,20 +795,25 @@ function resolve(schema, features)
       end
       if node.body then
          node.body = shallow_copy(node.body)
-         for k,v in pairs(node.body or {}) do
+         local extra = {}
+         for k,v in pairs(node.body) do
             if v.kind == 'uses' then
                -- Inline "grouping" into "uses".
                local grouping = lookup_lazy(env, 'groupings', v.id)
                node.body[k] = nil
                for k,v in pairs(grouping.body) do
-                  assert(not node.body[k], 'duplicate identifier: '..k)
-                  node.body[k] = v
+                  assert(not extra[k], 'duplicate identifier: '..k)
```

Comment:
If it's guaranteed grouping.body do not contain duplicated nodes, I can spare this assert.